From: Rob Evers <revers@redhat.com>
Date: Wed, 27 Oct 2010 15:23:00 -0400
Subject: [scsi] lpfc: fix lpfc_initial_flogi return on failure
Message-id: <1288192999-24221-9-git-send-email-revers@redhat.com>
Patchwork-id: 28933
O-Subject: [RHEL5.6 PATCH 08/27] lpfc: Fixed lpfc_initial_flogi not returning
	failure in one of failure condition
Bugzilla: 639028

lpfc: Fixed lpfc_initial_flogi not returning failure in one of failure condition

From: Rob Evers on behalf of Emulex <revers@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=636289

diff --git a/drivers/scsi/lpfc/lpfc_els.c b/drivers/scsi/lpfc/lpfc_els.c
index 6a8ff16..b450c73 100644
--- a/drivers/scsi/lpfc/lpfc_els.c
+++ b/drivers/scsi/lpfc/lpfc_els.c
@@ -1202,12 +1202,13 @@ lpfc_initial_flogi(struct lpfc_vport *vport)
 			return 0;
 	}
 
-	if (lpfc_issue_els_flogi(vport, ndlp, 0))
+	if (lpfc_issue_els_flogi(vport, ndlp, 0)) {
 		/* This decrement of reference count to node shall kick off
 		 * the release of the node.
 		 */
 		lpfc_nlp_put(ndlp);
-
+		return 0;
+	}
 	return 1;
 }
 
